doc/how.rst

```diff
@@ -3,87 +3,160 @@
 
-A benchmark consists in three elements: an objective function,
-a list of datasets, and a list of solvers.
+.. note::
 
+    The simplest way to create a benchmark is to copy an existing folder and
+    modify the element to fit the new structure.
+
+
+.. _objective:
 
 1. Objective
 ------------
 
-The objective function is defined through a Python class that
-allows to evaluate the function that should be minimized by solvers.
-A objective should define a `set_data` method that allows
-to specify the data and a `compute` method that allows
-to evaluate the objective for a given value of the iterate.
+The **objective function** is defined through a Python class.
+This class allows to evaluate the objective function to be minimized
+by the solvers. An objective class should define 3 methods:
+
+  - **set_data**: it allows to specify the data. See the data as a dictionary
+    of Python variables without any constraint.
+  - **compute**: it allows to evaluate the objective for a given value
+    of the iterate. This method should take only one parameter.
+  - **to_dict**: method that returns a dictionary to be passed
+    to the **set_objective** methods of solvers_.
+
+An objective class also needs to inherit from a base class called `BaseObjective`.
 
 Example
 ~~~~~~~
 
 .. literalinclude:: ../benchopt/tests/test_benchmarks/dummy_benchmark/objective.py
 
+.. _datasets:
+
 2. Datasets
 -----------
 
-A dataset defines what can be passed to the `set_data` method
-of an objective. A dataset should implement a `get_data` method
-whose output can be passed to the `set_data` of an objective.
+A dataset defines what can be passed to an objective. More specifically,
+a dataset should implement one method:
 
-Example
-~~~~~~~
+   - **get_data**: A method whose output consists of two things. First it outputs
+     the dimension of the optization problem (size of the iterates). Second it
+     outputs a dictionary that can be passed as `**kwargs` to
+     the **set_data** method of an objective_.
+
+A dataset class also needs to inherit from a base class called `BaseDataset`.
 
-Using a real dataset:
+Example using a real dataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. literalinclude:: ../benchopt/tests/test_benchmarks/dummy_benchmark/datasets/boston.py
 
-You can also define a parametrized dataset for example to test
+You can also define a parametrized simulated dataset for example to test
 across problem dimensions.
 
-Example of parametrized dataset
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Example of parametrized simulated dataset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Using simulated data:
+Sometimes one wants to test the solvers for variants of the same dataset.
+For example, one may want to change the dataset size, the noise level etc.
+To be able to specify parameters to get a dataset you can use a class
+attribute called `parameters`. This parameter must be a dictionary
+whose keys can be passed to the `__init__` of the class. The benchopt
```

Comment:
The > Then ?

Reply:
This is actually not necessary to be able to pass them to the class `__init__`. They will be stored anyway as `self.key = value`